packages/cosmos_ui_components/lib/components/cosmos_text_field.dart

```diff
@@ -42,17 +49,33 @@ class _CosmosTextFieldState extends State<CosmosTextField> {
       controller: controller,
       maxLines: widget.maxLines,
       maxLength: widget.maxLength,
-      onChanged: widget.onChanged,
+      keyboardType: widget.keyboardType,
+      onChanged: (value) {
+        widget.onChanged(value);
+        setState(() {});
+      },
       decoration: InputDecoration(
         border: UnderlineInputBorder(borderSide: BorderSide(color: theme.colors.inputBorder)),
+        hintText: widget.hint,
+        hintStyle: CosmosTextTheme.copy0Normal,
         counterText: '',
-        suffix: InkWell(
-          onTap: () {
-            controller.clear();
-            widget.onChanged('');
-          },
-          child: SizedBox(height: 17, width: 17, child: Image.asset('assets/images/cross.png', package: packageName)),
-        ),
+        suffixIcon: controller.text.isNotEmpty ? null : widget.suffix,
+        suffix: widget.suffix != null ? (controller.text.isNotEmpty ? _buildClearButton() : null) : _buildClearButton(),
```

Comment:
the general rule of if/else statements is that the condition inside if statement should be non-negated in order to avoid double negation, i.e:

instead of writing
```
if(false) {
  "its false"
} else {
"its not false" => "its true'
}
```

do this:

```
if(true) {
   "true!"
 } else {
   "false"
}
```

this is because double negation is harder to comprehend than just a true statement

Reply:
Updated